pkg/telemetry/consumer.go

```diff
@@ -56,12 +56,69 @@ func NewConsumer(s Serializer, f Forwarder) *consumer {
 	}
 }
 
+// Intake returns a channel on which this consumer will wait for data to consume it.
 func (c *consumer) Intake() chan<- types.Report {
 	return c.ch
 }
 
+// Close closes the consumer.
 func (c *consumer) Close() {
 	c.once.Do(func() {
 		close(c.done)
 	})
 }
+
+type rawconsumer struct {
+	logger logr.Logger
+	once   sync.Once
+	ch     chan types.Report
+	done   chan struct{}
+}
+
+// RawForwarder is used to forward raw, unserialized telemetry reports to configured
+// destination(s).
+type RawForwarder interface {
+	Name() string
+	Forward(types.Report) error
+}
+
+// NewRawConsumer creates a new rawconsumer that will use the provided raw forwarder
+// to forward received reports.
+func NewRawConsumer(f RawForwarder) *rawconsumer {
+	var (
+		ch     = make(chan types.Report)
+		done   = make(chan struct{})
+		logger = defaultLogger() // TODO: allow configuration
```

Comment:
Do we have an issue to follow up on this, or if not can we make one?

Reply:
Added: https://github.com/Kong/kubernetes-telemetry/issues/46 and amended the comment.